File: internal/lang/node/templates/_cloudcc_runtime/secret.js
// Secret backed by AWS Secrets Manager.

import { GetSecretValueCommand, PutSecretValueCommand, SecretsManagerClient } from "@aws-sdk/client-secrets-manager";

import { common, env, slug } from "./client.js";

export function connect(id) {
  const arn = env(`CLOUDCC_SECRET_${slug(id)}_ARN`, "persistSecret", id);
  return new Secret(id, arn, new SecretsManagerClient(common()));
}

export class Secret {
  constructor(id, arn, client) {
    this.id = id;
    this._arn = arn;
    this._client = client;
  }

  async get() {
    const out = await this._client.send(new GetSecretValueCommand({ SecretId: this._arn }));
    return out.SecretString ?? Buffer.from(out.SecretBinary).toString("utf8");
  }

  async set(value) {
    await this._client.send(
      new PutSecretValueCommand({ SecretId: this._arn, SecretString: String(value) }),
    );
  }
}

File: internal/cli/testdata/golden/mixed/worker/_cloudcc_runtime/secret.py
"""Secret backed by AWS Secrets Manager."""

from . import _client


def connect(id):
    """Return a client for the secret declared as ``persist_secret(id)``."""
    arn = _client.env("CLOUDCC_SECRET_%s_ARN" % _client.slug(id), "persist", id)
    return Secret(id, arn, _client.client("secretsmanager"))


class Secret:
    def __init__(self, id, arn, sm):
        self.id = id
        self._arn = arn
        self._sm = sm

    def get(self):
        """Return the secret's current value."""
        response = self._sm.get_secret_value(SecretId=self._arn)
        if "SecretString" in response:
            return response["SecretString"]
        return response["SecretBinary"].decode("utf-8")

    def set(self, value):
        """Replace the secret's value."""
        self._sm.put_secret_value(SecretId=self._arn, SecretString=str(value))

    def __repr__(self):
        return "<Secret %r (secretsmanager)>" % self.id
